src/toil/test/jobStores/jobStoreTest.py

```diff
@@ -1276,19 +1277,7 @@ def testSDBDomainsDeletedOnFailedJobstoreBucketCreation(self):
             else:
                 self.fail()
             finally:
-                try:
-                    for attempt in retry_s3():
-                        with attempt:
-                            s3.delete_bucket(bucket=bucket)
-                except S3ResponseError as e:
-                    # The actual HTTP code of the error is in status.
-                    # See https://github.com/boto/boto/blob/91ba037e54ef521c379263b0ac769c66182527d7/boto/exception.py#L77-L80
-                    # See also: https://github.com/boto/boto/blob/91ba037e54ef521c379263b0ac769c66182527d7/boto/exception.py#L154-L156
-                    if e.status == 404:
-                        # The bucket doesn't exist; maybe a failed delete actually succeeded.
-                        pass
-                    else:
-                        raise
+                client.delete_bucket(Bucket=bucketName)
```

Comment:
This is now safe when the bucket is gone already?

Reply:
Good catch. It seems like botocore does throw an error if the bucket doesn't exist, which wasn't obvious from the documentation. I shall create an update along with a fix for the `JobFileStoreTest.testJobFileStore` travis test failure.